ts/components/ui/IOScrollView.tsx

```diff
@@ -0,0 +1,369 @@
+    () =>
+      !needSafeAreaMargin || excludeSafeAreaMargins
+        ? IOVisualCostants.appMarginDefault
+        : safeAreaMargin,
+    [needSafeAreaMargin, excludeSafeAreaMargins, safeAreaMargin]
+  );
+
+  /* GENERATE EASING GRADIENT
+     Background color should be app main background
+     (both light and dark themes) */
+  const HEADER_BG_COLOR: ColorValue = IOColors[theme["appBackground-primary"]];
+
+  const { colors, locations } = easeGradient({
+    colorStops: {
+      0: { color: hexToRgba(HEADER_BG_COLOR, 0) },
+      1: { color: HEADER_BG_COLOR }
+    },
+    easing: Easing.ease,
+    extraColorStopsPerTransition: 20
+  });
+
+  /* When the secondary action is visible, add extra margin
+     to avoid little space from iPhone bottom handle */
+  const extraBottomMargin: number = useMemo(
+    () => (secondaryAction && needSafeAreaMargin ? extraSafeAreaMargin : 0),
+    [needSafeAreaMargin, secondaryAction]
+  );
+
+  /* Safe background block. Cover at least 85% of the space
+     to avoid glitchy elements underneath */
+  const safeBackgroundBlockHeight: number = useMemo(
+    () => (bottomMargin + actionBlockHeight) * 0.85,
+    [actionBlockHeight, bottomMargin]
+  );
+
+  /* Total height of "Actions + Gradient" area */
+  const gradientAreaHeight: number = useMemo(
+    () => bottomMargin + actionBlockHeight + gradientSafeAreaHeight,
+    [actionBlockHeight, bottomMargin]
+  );
+
+  /* Height of the safe bottom area, applied to the ScrollView:
+     Actions + Content end margin */
+  const safeBottomAreaHeight: number = useMemo(
+    () => bottomMargin + actionBlockHeight + contentEndMargin,
+    [actionBlockHeight, bottomMargin]
+  );
+
+  const handleScroll = useAnimatedScrollHandler(
+    ({ contentOffset, layoutMeasurement, contentSize }) => {
+      const scrollPosition = contentOffset.y;
+      const maxScrollHeight = contentSize.height - layoutMeasurement.height;
+      const scrollPercentage = scrollPosition / maxScrollHeight;
+
+      // eslint-disable-next-line functional/immutable-data
+      scrollPositionAbsolute.value = scrollPosition;
+      // eslint-disable-next-line functional/immutable-data
+      scrollPositionPercentage.value = scrollPercentage;
+    }
+  );
+
+  const opacityTransition = useAnimatedStyle(() => ({
+    opacity: interpolate(
+      scrollPositionPercentage.value,
+      [0, gradientOpacityScrollTrigger, 1],
+      [1, 1, 0],
+      Extrapolate.CLAMP
+    )
+  }));
+
+  /* Set custom header with `react-navigation` library using
+     `useLayoutEffect` hook */
+
+  const scrollValues: IOSCrollViewHeaderScrollValues = useMemo(
+    () => ({
+      contentOffsetY: scrollPositionAbsolute,
+      triggerOffset: snapOffset || 0
+    }),
+    [scrollPositionAbsolute, snapOffset]
+  );
+
+  useLayoutEffect(() => {
+    if (headerConfig) {
+      navigation.setOptions({
+        header: () => (
+          <HeaderSecondLevel {...headerConfig} scrollValues={scrollValues} />
+        )
+      });
```

Comment:
```suggestion
        ),
        headerTransparent: headerConfig.transparent
      });
```

if we don't propagate properly this behaviour the transparent option won't work properly